include/service/entities/site_query_service.js

```diff
@@ -0,0 +1,208 @@
+      }
+    }
+    return where;
+  }
+
+  function modifyLoadOptions(site, options) {
+    if (pb.config.multisite.enabled) {
+      var target = _.clone(options);
+
+      target.where = target.where || {};
+      target.where = modifyLoadWhere(site, target.where);
+      return target;
+    }
+    // else do nothing
+    return options;
+  }
+
+  function addToOr(whereClause, conditions) {
+    if ('$or' in whereClause) {
+      var orClause = whereClause.$or;
+      addToAnd(whereClause, [{$or: orClause}, {$or: conditions}]);
+      delete whereClause.$or;
+    } else {
+      whereClause.$or = conditions;
+    }
+  }
+
+  function addToAnd(whereClause, conditions) {
+    if ('$and' in whereClause) {
+      var andClause = whereClause.$and;
+      andClause.push.apply(andClause, conditions);
+    } else {
+      whereClause.$and = conditions;
+    }
+  }
+
+  function applySiteOperation(self, callback, delegate) {
+    if (siteSpecific(self)) {
+      delegate(self.siteUId, callback);
+    } else {
+      delegate(self.siteUId, function (err, cursor) {
+        if (util.isError(err)) {
+          callback(err, cursor);
+        } else {
+          cursor.count(function (countError, count) {
+            if (util.isError(countError)) {
+              callback(countError);
+            } else if (count) {
+              callback(err, cursor);
+            } else {
+              delegate(GLOBAL_SITE, callback);
+            }
+          });
+        }
+      })
+    }
+  }
+
+  function siteSpecific(self) {
+    return self.onlyThisSite || isGlobal(self.siteUId);
+  }
+
+  function isGlobal(siteUId) {
+    return !siteUId || siteUId === GLOBAL_SITE;
+  }
+
+  function modifySave(site, objectToSave) {
+    if (pb.config.multisite && !(SITE_FIELD in objectToSave)) {
+      objectToSave[SITE_FIELD] = site;
+    }
+    // else do nothing
+    return objectToSave;
+  }
+
+  /**
+   * Overriding protected method of DAO to achieve site-aware query
+   * @override
+   * @protected
+   * @method _doQuery
+   * @param {Object} options
+   * @param {Function} callback
+   */
+  SiteQueryService.prototype._doQuery = function (options, callback) {
+    var self = this;
+    applySiteOperation(self, callback, function (site, opCallback) {
+      var moddedOptions = modifyLoadOptions(site, options);
+      DAO.prototype._doQuery.call(self, moddedOptions, opCallback);
+    });
+  };
+
+  /**
+   * Wrapper for site-aware DAO.save.  Saves object to database
+   *
+   * @param dbObj
+   * @param options
+   * @param callback
+   */
+  SiteQueryService.prototype.save = function (dbObj, options, callback) {
+    dbObj = modifySave(this.siteUId, dbObj);
+    DAO.prototype.save.call(this, dbObj, options, callback);
+  };
+
+  /**
+   * Gets all collection names
+   * @param {Function} cb
+   */
+  SiteQueryService.prototype.getCollections = function (cb) {
+    var dao = new pb.DAO();
+    dao.listCollections({}, function(err, items) {
```

Comment:
if SiteQueryService inherits from DAO why do you need a new instance here to call ```listCollections```?  Can you not just do:
```
SiteQueryService.prototype.getCollections = function (cb) {
  this.listCollections({}, cb);
};
```